src/array/mod.rs

```diff
@@ -234,7 +243,8 @@ macro_rules! impl_from {
             /// Implement `AbcArray -> ArrayImpl`
             impl From<$AbcArray> for ArrayImpl {
                 fn from(array: $AbcArray) -> Self {
-                    Self::$Abc(array)
+                    // Self::$Abc(array)
```

Comment:
Can we `Self::$Abc(Arc::new(array))` here? Then no need to add `new_bool`

Reply:
`Self::$Abc(array.into())` might be better. By using `.into()`, we can have both `Arc` and the original representation in such intermediate state.

Reply:
Great suggestion, thanks!